### ISSUE TITLE
"TypeError: innerDecoder is not a function" prevents querying or transacting on chains.

### ISSUE THREAD

**Issue by 71walceli** [tien/reactive-dot#405]
**After updating `reactive-dot` and `polkadot-api`, I'm encountering an error that prevents me from using any API calls for queries or transactions on any chain, including local Rococo instances.**

**Error Details:**

* **Error Type:** `TypeError: innerDecoder is not a function`
* **Additional Information:**
    * Minified code makes debugging difficult.
    * Unsure if the issue originates from `polkadot-api` or `reactive-dot`.
    * Potential causes:
        * Update incompatibility
        * Error in my implementation (using lightClient as per updated examples that used smoldot worker)

Full error message:
```
TypeError: innerDecoder is not a function
    at chunk-JGGR4AA4.js?v=db58212e:318:14
    at chunk-JGGR4AA4.js?v=db58212e:109:43
    at chunk-JGGR4AA4.js?v=db58212e:645:28
    at chunk-JGGR4AA4.js?v=db58212e:349:90
    at Array.map (<anonymous>)
    at chunk-JGGR4AA4.js?v=db58212e:349:73
    at chunk-JGGR4AA4.js?v=db58212e:109:43
    at chunk-JGGR4AA4.js?v=db58212e:141:61
    at chunk-JGGR4AA4.js?v=db58212e:5358:17
    at chunk-JGGR4AA4.js?v=db58212e:5097:44
```
![image](https://github.com/user-attachments/assets/b8d9adca-4b83-47c9-b3ab-0642e744df5b)

Please let me know if you need any extra info.

**Comment by 71walceli** [comment]
Code samples from the project I'm working on:

config.ts
```
const getProviders = () => {
  const lightClientProvider = createLightClientProvider();
  const polkadot = lightClientProvider.addRelayChain({ id: "polkadot" });
  const kusama = lightClientProvider.addRelayChain({ id: "kusama" });
  const westend = lightClientProvider.addRelayChain({ id: "westend" });

  return {
    lightClientProvider,
    polkadot,
    kusama, 
    westend
  };};
export let providers = getProviders();

export const config = {
chains: {
    people_polkadot: {
      name: "Polkadot",
      descriptor: people_polkadot,
      provider: providers.polkadot.addParachain({ id: "polkadot_people" }),
      registrarIndex: import.meta.env.VITE_APP_REGISTRAR_INDEX__PEOPLE_POLKADOT,
    },
    polkadot: {
      name: "Polkadot",
      descriptor: polkadot,
      provider: providers.polkadot,
    },
    // ...
}
```

App.tsx
```
const typedApi = useTypedApi({ chainId })
const getIdAndJudgement = useCallback(() => typedApi.query.Identity.IdentityOf
    .getValue(accountStore.address)
    .then((result) => {
      // Set up state to render identity data
    })
    .catch(e => {
      if (import.meta.env.DEV) {
        console.error("Couldn't get identityOf");
        console.error(e);
      }
    })
  , [address, typedApi]);
```

**Comment by tien** [comment]
The error appears to come from `scale-ts`, which is a dependency of `polkadot-api`.

@71walceli can you first try to regenerate the descriptors using `npx papi generate`?

**Comment by 71walceli** [comment]
Let's see... I tried updating metadata several times already. 

```
...
.papi/descriptors/src/westend.ts(2522,8): error TS2314: Generic type 'StorageDescriptor' requires 4 type argument(s).
.papi/descriptors/src/westend.ts(2528,19): error TS2314: Generic type 'StorageDescriptor' requires 4 type argument(s).
.papi/descriptors/src/westend.ts(2536,23): error TS2314: Generic type 'StorageDescriptor' requires 4 type argument(s).

Compilation done with 1666 errors
bun install
[3.61ms] ".env"
bun install v1.1.30 (7996d06b)

+ @polkadot-api/descriptors@.papi/descriptors

1 package installed [421.00ms]
```

seems it got installed, but those errors started to pop up. I cleared node_modules, but still get same result. Might need to nuke .papi/descriptors...

**Comment by 71walceli** [comment]
that happens with `papi update` as well...

**Comment by 71walceli** [comment]
> The error appears to come from `scale-ts`, which is a dependency of `polkadot-api`.

Hmmmmm....

**Comment by tien** [comment]
Probably a bug with `polkadot-api` if those commands break 🤔.

**Comment by tien** [comment]
Check if all papi related dependencies are up-to-date `@polkadot-api/*`

**Comment by 71walceli** [comment]
yeah maybe. Tested with `polkadot-api` `1.7.7` to `1.7.8`... Please let me check if there are similar issues.

**Comment by 71walceli** [comment]
> Check if all papi related dependencies are up-to-date `@polkadot-api/*`

I did `bun update`, which updated all deps. But that produced some errors with `postcss` init when launckhing dev container.  So I reverted the commit, picking and cloosing packages to revert. and that seems to have fixed it. Best part is that the app I'm working on will now fully function and be able to tonnect to Kusama chain as per my testing so far, as well as local one.
